lib/utils/compose_ts.ts

```diff
@@ -1355,15 +1347,12 @@ export async function deployProject(
 			);
 		}
 	} finally {
-		runloop = runSpinner(tty, spinner, `${prefix}Saving release...`);
-		release.end_timestamp = new Date();
-		if (release.id != null) {
-			try {
+		await runSpinner(tty, spinner, `${prefix}Saving release...`, async () => {
+			release.end_timestamp = new Date();
+			if (release.id != null) {
 				await releaseMod.updateRelease(pineClient, release.id, release);
-			} finally {
-				runloop.end();
```

Comment:
This also fixes a bug where if `release.id == null` then we never actually hit this specific `runloop.end` and so would continue spinning forever - I don't know if that ever happens in practice but the callback api makes it far easier to ensure the spinner is always ended and avoids the potential for the issue